### ISSUE TITLE
Better support when language server doesn't work

### ISSUE THREAD

**Issue by angelozerr** [eclipse/lsp4e#748]
LSP4E provides a great support for language server when it can start and when everything is working well, but when language server crashes or cannot be started,it is a very nightmare, because in short any operations tries again and again to start the language server and it freezes the Eclipse IDE. 

Here a sample issue https://github.com/jbosstools/jbosstools-quarkus/issues/235

In short, there is a problem with the Quarkus language server and there are several problems:

 * the first is that it freezes on each operations and Java editor (Quarkus language server is mapped to Java content type) freezes for any operations like hyperlink. The only solution for the user is to uninstall the plugin although the plugin provides another features than language server.
 * we have none overview about the problem (the language server can be sarted, on which operations is faiiling?)

On LSP4IJ (LSP support for IJ), I managed this usecase by disabling automaticly the language server (after 50 attempt of language server start). The LSP console shows that the language server is disabled and show the error stack trace. As we can see the server status, we can know that the problem comes from the start of language server. User can copy the start command to try it a shell command to investigate more the problem (One of problem for instance is that it starts the language server with a wron Java version runtime)

Here a demo with a language server which cannot be started:

![LSPCrashOnStart](https://github.com/eclipse/lsp4e/assets/1932211/5e33e262-3e6b-443f-8cab-34bfab2dd7a4)

In this demo you can see:

 * stack trace error on LSP console
 * the server status which shows that after several attemps become disabled

User can choose too to disable it with Stop button

![image](https://github.com/eclipse/lsp4e/assets/1932211/20d7f9fc-2a49-4594-a165-3d5c910cd261)

In other words:

 * there are no freeze if language server cannot start
 * user can choose to disable it (with LSP4E you have already implemented that by unchecking the language server)
 * LSP console shows the error message easily

**Comment by mickaelistria** [comment]
Not performing more than N restarts (20, 50, ...) upon startup failure would be a welcome addition.
